src/main.rs

```diff
@@ -49,6 +49,23 @@ ow=0:or=0;38;5;16;48;5;203:no=0:ex=1;38;5;203:cd=0;38;5;203;48;5;236:mi=0;38;5;1
 ";
 
+/// Default value for the path_separator, mainly for MSYS/MSYS2, which set the MSYSTEM
+/// environment variable, and we set fd's path separator to '/' rather than Rust's default of '\'.
+///
+/// Returns Some to use a nonstandard path separator, or None to use rust's default on the target
+/// platform.
+fn default_path_separator() -> Option<String> {
+    if cfg!(windows) {
+        let msystem = env::var("MSYSTEM").ok()?;
+        match msystem.as_str() {
+            "MINGW64" | "MINGW32" | "MSYS" => Some("/".to_owned()),
+            _ => None,
+        }
+    } else {
+        None
+    }
+}
```

Comment:
Maybe we could move this function to `src/filesystem.rs`?

Reply:
done